src/components/app.jsx

```diff
@@ -33,21 +33,32 @@ const PrivateContext = () => (
 
 const PublicContext = withClient(({ client }) => {
   const [sharedDocumentId, setSharedDocumentId] = useState(null)
+  const [readOnly, setReadOnly] = useState(false)
```

Comment:
This means that for some time dependning on the network, the document can be read only when the user does not have the right permissions. What happens in this case if there are modifications ? Is this handled ?

Reply:
The changes will be refused by the server anyways and will be erased when the readOnly attribute will kick in. There should be a really bad timing for the editor to be ready before the readOnly being set but you're right, I'll chang the order  of a few lines below  to make sure it will never happen

Reply:
> What happens in this case if there are modifications ? Is this handled ?

If the editor is set readOnly after its first use, any modifications will be refused by the server anyways and will stay pending in the local editor.
When the editor goes into readonly, all these local changes will be deleted to only leave the server version.

> This means that for some time dependning on the network, the document can be read only when the user does not have the right permissions

Nope, the setReadOnly is always called before the setSharedDocumentId so we never initialize the editor prior to setting to readonly attribute.